ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CommandTracker.scala

```diff
@@ -222,11 +223,28 @@ private[commands] class CommandTracker[Context](
             s"A command with id $commandId is already being tracked. CommandIds submitted to the CommandTracker must be unique."
           ) with NoStackTrace
         }
-        val commandTimeout =
-          timeoutFromDeduplicationPeriod(commands.deduplicationPeriod, maximumCommandTimeout)
+        val commandTimeout = submission.value.timeout match {
+          // The command submission timeout takes precedence.
+          case Some(timeout) => durationOrdering.min(timeout, maximumCommandTimeout)
+          case None =>
+            commands.deduplicationPeriod match {
+              // We keep supporting the `deduplication_time` field as the command timeout,
+              // for historical reasons.
+              case Commands.DeduplicationPeriod.DeduplicationTime(deduplicationTimeProto) =>
+                val deduplicationTime = Duration.ofSeconds(
+                  deduplicationTimeProto.seconds,
+                  deduplicationTimeProto.nanos.toLong,
+                )
+                durationOrdering.min(deduplicationTime, maximumCommandTimeout)
+              // All other deduplication periods were added recently and should not influence the
+              // command timeout.
```

Comment:
Can you remove the time reference from this comment, please? I.e., 'recently' won't mean the same in a few months.
```suggestion
              // All other deduplication period types should not influence the command timeout.
```

Reply:
Done. Thanks for catching this.